### PR TITLE
avoid descheduling when ReplicaSet=1

### PR DESCRIPTION
**Pull request by jiangxiaobin96**
fix [#786](https://github.com/kubernetes-sigs/descheduler/issues/786)
Compared with setting `PodDisruptionBudget` for every RC/RS, easy to avoid evicting pods with `replicasSet=1`.

### CODE REVIEW

File: kubernetes/deployment/deployment.yaml [hunk 1]
```diff
@@ -19,7 +19,7 @@ spec:
       serviceAccountName: descheduler-sa
       containers:
         - name: descheduler
-          image: registry.k8s.io/descheduler/descheduler:v0.25.1
+          image: xiaobinimage/descheduler:v3
```
Comment (JaneLiuL) [inline kubernetes/deployment/deployment.yaml:22]: may i ask why change the image repo?

File: pkg/descheduler/evictions/evictions.go [hunk 2]
```diff
@@ -30,12 +30,14 @@ import (
 	"sigs.k8s.io/descheduler/metrics"
 
 	eutils "sigs.k8s.io/descheduler/pkg/descheduler/evictions/utils"
+	"sigs.k8s.io/descheduler/pkg/utils"
 )
 
 // nodePodEvictedCount keeps count of pods evicted on node
 type (
 	nodePodEvictedCount    map[string]uint
 	namespacePodEvictCount map[string]uint
```
Comment (JaneLiuL) [inline pkg/descheduler/evictions/evictions.go:39]: could you make the lines align?

**Review by JaneLiuL — COMMENTED** [review]
/ok-to-test

File: kubernetes/deployment/deployment.yaml (hunk 1 above)
Comment (jiangxiaobin96) [reply kubernetes/deployment/deployment.yaml:22]: Sorry, forget to change test env back. I will update soon.

File: pkg/descheduler/evictions/evictions.go (hunk 2 above)
Comment (jiangxiaobin96) [reply pkg/descheduler/evictions/evictions.go]: done

**Comment by AllenZMC** [comment]
@jiangxiaobin96  Excuse me, how is it going now?

**Comment by jiangxiaobin96** [comment]
> @jiangxiaobin96 Excuse me, how is it going now?

Let me solve the conflict and rebase.

**Comment by jiangxiaobin96** [comment]
Because this issue has been create some time ago, it has large difference with master branch. I will close this issue first and reconstruct this issue based on new branch.

**Comment by AllenZMC** [comment]
> Because this issue has been create some time ago, it has large difference with master branch. I will close this issue first and reconstruct this issue based on new branch.

How is it going now?

**Comment by jiangxiaobin96** [comment]
> > Because this issue has been create some time ago, it has large difference with master branch. I will close this issue first and reconstruct this issue based on new branch.
> 
> How is it going now?

testing, this week can finish

**Comment by jiangxiaobin96** [comment]
reconstruct  in [#1152 ](https://github.com/kubernetes-sigs/descheduler/pull/1152)

**Comment by damemi** [comment]
/reopen
keeping this open until the PR merges

**Pull request closed without merging**